custom_components/audiconnect/audi_services.py

```diff
@@ -614,20 +612,21 @@ async def check_request_succeeded(
 
         raise Exception("Cannot {action}, operation timed out".format(action=action))
 
-    # TR/2022-06-15: New secrect for X_QMAuth
+    # TR/2022-12-20: New secrect for X_QMAuth
     def _calculate_X_QMAuth(self):
         # Calcualte X-QMAuth value
         gmtime_100sec = int(
             (datetime.utcnow() - datetime(1970, 1, 1)).total_seconds() / 100
         )
-        xqmauth_secret = bytes([256-28,120,102,55,256-114,256-16,101,256-116,256-25,93,113,0,122,256-128,256-97,52,97,107,256-106,53,256-30,256-20,34,256-126,69,120,76,31,99,256-24,256-115,6])
```

Comment:
Could you give a clue where to find that xqmauth_secret array of bytes in original application? I was looking a few days ago at the issue when app stoped to but my current knowledge about that is not enought to fix that. Is is somewhere in the code and it is required to decomplied original APK or is it just some kind of encoding of clientId OAuth parameter?

Reply:
I patched the myAudi apk and observed the runtime behavior and values during a login procedure with the Frida tool.
In theory it would be possible to get the values from the decompiled app, but runtime analysis was easier.